docker_builds/Dockerfile

```diff
@@ -0,0 +1,70 @@
+FROM fedora:26
+
+RUN dnf update -y && dnf install -y --best --allowerasing \
+    bash \
+    boost \
+    boost-devel \
+    cmake \
+    double-conversion \
+    double-conversion-devel \
+    gcc-c++ \
+    python \
+    make \
+    protobuf \
+    protobuf-devel \
+    protobuf-python \
+    log4cxx \
+    log4cxx-devel \
+    yaml-cpp \
+    yaml-cpp-devel \
+    wget \
+    which \
+    git \
+    autoconf \
+    automake \
+    gflags-devel \
+    glog-devel \
+    libtool \
+    openssl-devel \
+    libevent-devel \
+    python \
+    python-devel \
+    python-pip \
+    python-virtualenvwrapper \
+    python3 \
+    python3-virtualenv \
+    sudo \
+    libuv-devel && \
```

Comment:
Is this stuff all lumped in here to save on more lines in this Dockerfile? It's certainly ugly in this format

Reply:
Its better for docker's caching IIRC

Reply:
that's what I figured, but it's so ugly I have to wonder if 1/2 more lines to copy a script then run it is worth it

Reply:
I did this to try and avoid intermediate layers since it seemed like dockerhub was deleting them.  It also made the final image a lot smaller because /var/tmp was filled with build artifacts that we didn't need and didn't want in the intermediate layers.

Reply:
Works for me, not a huge issue